src/main/java/org/openlmis/stockmanagement/service/StockCardAggregate.java

```diff
@@ -122,26 +126,33 @@ public Long getStockoutDays(LocalDate startDate, LocalDate endDate) {
 
     long stockOutDays;
     if (startDate == null || endDate == null) {
-      stockOutDays = calculateStockoutDays(
-          getStockoutPeriods(stockOnHands, endDate), startDate, endDate);
+      stockOutDays = calculateStockoutDays(getStockoutPeriods(stockOnHands, endDate),
+              startDate, endDate);
     } else {
 
-      int sumOfStockOnHandDuringPeriod = 0;
+      boolean wasAnySohChangesInPeriod = false;
       for (Map.Entry<LocalDate, Integer> entry : stockOnHands.entrySet()) {
         if (isAfterOrEqual(startDate, entry.getKey()) && isBeforeOrEqual(endDate, entry.getKey())) {
-          sumOfStockOnHandDuringPeriod += entry.getValue();
+          wasAnySohChangesInPeriod = true;
+          break;
         }
       }
-      if (sumOfStockOnHandDuringPeriod == 0) {
-        long daysBetween = DAYS.between(startDate, endDate) + 1;
-        // According to OLMIS project specification month length can be maximum 30 days.
-        if (daysBetween >= 28) {
-          daysBetween -= 1;
+
+      int beginningBalance = getMostRecentStockOnHandBeforeDate(stockOnHands, startDate);
+      if (!wasAnySohChangesInPeriod) {
+        if (beginningBalance > 0) {
+          stockOutDays = 0L;
+        } else {
+          long daysBetween = DAYS.between(startDate, endDate) + 1;
+          // According to OLMIS project specification month length can be maximum 30 days.
```

Comment:
Hmm, comment says maximum 30 days, but an 'if' below doesn't check max - what is the reason for this comment? Is it some leftover from previous (even before your changes) solution?